xgcm/grid.py

```diff
@@ -21,7 +21,7 @@
 import xarray as xr
 from dask.array import Array as Dask_Array
 
-from . import comodo, gridops
+from . import comodo, gridops, sgrid
```

Comment:
I think we should move both comodo and sgrid into a new module `metadata_parsers.py`?

Reply:
Oh actullay i misunderstood the way this was setup. I wonder if it might be better to define an Sgrid class in a dedicated module? Maybe @TomNicholas can chime in here.

Reply:
I'd say there is certainly potential for a `gridparsers` module or similar that holds sgrid, comodo, and any other future additions. Alternatively perhaps a `metadata_parsers` module if we want to call cf_parsing from inside sgrid or comodo parsing (see https://github.com/xgcm/xgcm/issues/566#issuecomment-1410690075)

Reply:
A separate module is definitely the direction I would like to see things go in. The logical flow inside there might have to incorporate both CF and sgrid logic, but we want all of that to be clearly separate from just creating a `Grid` object from a given set of kwargs.

Reply:
@TomNicholas so if I understand correctly a user would need to call `kwargs = metadata_parsers.parse_sgrid(ds)` (or similar for comodo etc) before then calling `da_grid = grid.Grid(kwargs)`?

Or do you want a generic `kwargs = metadata_parsers.parse_metadata()` method in there that then cycles through the various options (sgrid, cf, comodo etc) to try and auto-infer?

Reply:
> Or do you want a generic kwargs = metadata_parsers.parse_metadata() method in there that then cycles through the various options (sgrid, cf, comodo etc) to try and auto-infer?

Very good question, I think we should discuss this in the meeting.

Reply:
We can discuss the specifics of the function return values in the meeting, but I have made the requested change to the code so `grid.py` now has `from . import gridops, metadata_parsers` and does not refer directly to either `sgrid.py` or `comodo.py`.